# Profitable free apps to build for the Apple and Google Play store

This project is to uncover the best type of free mobile app to build that will attract a high number of users in the Apple and Google play stores. In order to maximise the revenue from in-app ads, we have to choose the right type of app to build. 

As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play.

The [Google Dataset](https://www.kaggle.com/lava18/google-play-store-apps]) can be downloaded directly via [this link](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv)

The [Apple Dataset](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) can be downloaded directly via [this link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv)

To begin, we'll open the two sets and explore the data in them

In [64]:
from csv import reader

In [65]:
### Opening the apple store data ###
OpenApple = open('AppleStore.csv', encoding="utf8")
ReadApple = reader(OpenApple)
AppleStoreData = list(ReadApple)
Appleheader = AppleStoreData[0]
ios = AppleStoreData[1:]

### Opening the Google store data set ###
OpenGoogle = open('googleplaystore.csv', encoding="utf8")
Readgoogle = reader(OpenGoogle)
google_store_data = list(Readgoogle)
googleheader = google_store_data[0]
android = google_store_data[1:]

We'll write a function to explore the data set

In [66]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))
    

In [67]:
print(Appleheader)
explore_data(ios, 0, 5, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 7197
Number of columns: 16


In [68]:
print(googleheader)
explore_data(android, 0, 4, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 10841
Number of columns: 13


The Apple data set has 7,197 rows & the Google data gas 10,841 rows.
The useful columns for the Apple store seem to be ID, track name, rating_count_total - these are described in the documentation

The Google store columns are more straight forward with App, Category, Ratings, Installs, Price and co

The Google Play data set has a dedicated discussion section, and we can see that one of the [discussions outlines](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015)  an error for row 10472.

In [69]:
print(android[10472])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


In [70]:
del android[10472]

**I saw someone suggested in the discussion room that it makes sense to figure out the category and add it to the list, instead of deleting it. since all other columns are perfectly fine. I might come back to do that.** 

In [71]:
android[10472]

['osmino Wi-Fi: free WiFi',
 'TOOLS',
 '4.2',
 '134203',
 '4.1M',
 '10,000,000+',
 'Free',
 '0',
 'Everyone',
 'Tools',
 'August 7, 2018',
 '6.06.14',
 '4.4 and up']

To ensure the quality of the data, I'm going to check if there are any duplicate entries. Using the code below, I'll seperate apps with one entry & apps with multiple entries below. 

In [72]:
#Checking for duplicate apps 
duplicate_apps = []
unique_apps = []

for app in android:
    app_name = app[0]
    if app_name in unique_apps:
        duplicate_apps.append(app_name)
    else:
        unique_apps.append(app_name)
        

In [73]:
print (len(duplicate_apps) , len(unique_apps))

1181 9659


In [74]:
duplicate_apps[0:20]  

['Quick PDF Scanner + OCR FREE',
 'Box',
 'Google My Business',
 'ZOOM Cloud Meetings',
 'join.me - Simple Meetings',
 'Box',
 'Zenefits',
 'Google Ads',
 'Google My Business',
 'Slack',
 'FreshBooks Classic',
 'Insightly CRM',
 'QuickBooks Accounting: Invoicing & Expenses',
 'HipChat - Chat Built for Teams',
 'Xero Accounting Software',
 'MailChimp - Email, Marketing Automation',
 'Crew - Free Messaging and Scheduling',
 'Asana: organize team projects',
 'Google Analytics',
 'AdWords Express']

We have 1181 entries of apps with duplicate reviews! 
We are going to have to filter them out and delete all but one entry. 
This will be carried out using a criterion that makes each entry unqiue. If they were all exactly the same, then I could write code to delete all but one entry. 
lets print some out some duplicates to see what they look like. 

In [75]:
for app in android:
    appname = app[0]
    if appname == 'Box':
      print (app)
    
if 'Box' in unique_apps:
    print('yes')

['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
yes


In [76]:
for app in android:
    appname = app[0]
    if appname == 'Instagram':
      print (app)
    

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


With some entries, the number of rating changes with each entry while some of them have the ratings stay the same. As this is the onlyy value that changes, we'll use this as a criterion to filter out the duplicates. We'll use two arguments, one that checks if theyre the same or higher ??

We're going to create a dictionary to do the deletion of duplicate values. 


In [77]:
reviews_max = {}   ###initialising a dictionary with key:value are app name:number of ratings. Recall that to access to 
                    ### a dictionary's value is dictionary[key]

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
        
        

In [78]:
print (len(reviews_max))

9659


to create a list of lists to continue anylysing the data, im going to create a list 'android_clean' that will contain unique entries. the second emplty list 'already added' is to ensure that the same app doesnt get added multiple times. the criteria name and number of reviews are important here. 

The dictionary 'reviews_max' already contains the entry with the maximum number of reviews. this is what we'll use to compare and extract each row into the android clean list. 

In [79]:
android_clean = []  #to store new cleaned data set 
already_added = []  # to store app names 


for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if n_reviews == reviews_max[name] and name not in already_added:
        
        android_clean.append(app)
        already_added.append(name)
        
        
print(len(android_clean))
# already_added[0:50]

9659


Great! we have a list with 9659 unique values. now we continue to clean


In [80]:
#Checking for duplicate apps in Apple store
duplicate_apps_apple = []
unique_apps_apple = []

for app in ios:
    app_id = float(app[0]) #always remember to change the numbers to float
    if app_id in unique_apps:
        duplicate_apps_apple.append(app_id)
    else:
        unique_apps_apple.append(app_id)
        
print (len(duplicate_apps_apple) , len(unique_apps_apple))
        

0 7197


there are no duplicates, so we move unto removing non-english characters

In [81]:
def eng_filter(string):
    non_eng = 0
    for character in string:
        number_ord = ord(character)
        if number_ord > 127:
            non_eng += 1 
            
    if non_eng > 3:
        return False
    elif non_eng <= 3 : 
        return True
        

In [82]:
eng_filter('Instgram')

True

In [83]:
eng_filter('爱奇艺PPS -《欢乐颂2》电视剧热播')

False

In [84]:
eng_filter('Docs To Go Free Office Suite')

True

In [85]:
eng_filter('Instachat')

True

In [86]:
#to check for non english names 
non_eng_app = []
for app in android_clean:
    name = app[0]
    if eng_filter(name) == False:
        non_eng_app.append(name)
    
    

In [87]:
len(non_eng_app) 


45

there are 45 non english apps in the google store data set


there are no non-english apps in the apple store data set

**im going to filter out the non-english apps and add them to android_clean_english**

In [88]:
android_eng = []
for app in android_clean:
    name = app[0]
    if name not in non_eng_app:
        android_eng.append(app)

In [89]:
len(android_eng)

9614

In [93]:
ios_eng = []
for app in ios:
    name = app[1]
    if eng_filter(name):
        ios_eng.append(app)
        
print('There are ', len(ios_eng), 'english apps in the Apple App store')

There are  6183 english apps in the Apple App store


In [94]:
googleheader

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

In [95]:
# android_eng[1000:1010]

In [96]:
free_android_apps = []
paid_apps = []
for app in android_eng:
    if_free = app[6]
    if if_free == 'free' or if_free == 'Free':
        free_android_apps.append(app)
    else:
        paid_apps.append(app)
        
print('The number of free apps in the google store data set is: ' + str(len(free_android_apps)), 
      'and the number of paid apps in the google store data set is: ' + str(len(paid_apps)))

The number of free apps in the google store data set is: 8863 and the number of paid apps in the google store data set is: 751


Lets see the data to make sure its properly done

In [98]:
 paid_apps[0:5]

[['TurboScan: scan documents and receipts in PDF',
  'BUSINESS',
  '4.7',
  '11442',
  '6.8M',
  '100,000+',
  'Paid',
  '$4.99',
  'Everyone',
  'Business',
  'March 25, 2018',
  '1.5.2',
  '4.0 and up'],
 ['Tiny Scanner Pro: PDF Doc Scan',
  'BUSINESS',
  '4.8',
  '10295',
  '39M',
  '100,000+',
  'Paid',
  '$4.99',
  'Everyone',
  'Business',
  'April 11, 2017',
  '3.4.6',
  '3.0 and up'],
 ['Puffin Browser Pro',
  'COMMUNICATION',
  '4.0',
  '18247',
  'Varies with device',
  '100,000+',
  'Paid',
  '$3.99',
  'Everyone',
  'Communication',
  'July 5, 2018',
  '7.5.3.20547',
  '4.1 and up'],
 ['Truth or Dare Pro',
  'DATING',
  'NaN',
  '0',
  '20M',
  '50+',
  'Paid',
  '$1.49',
  'Teen',
  'Dating',
  'September 1, 2017',
  '1.0',
  '4.0 and up'],
 ['Private Dating, Hide App- Blue for PrivacyHider',
  'DATING',
  'NaN',
  '0',
  '18k',
  '100+',
  'Paid',
  '$2.99',
  'Everyone',
  'Dating',
  'July 25, 2017',
  '1.0.1',
  '4.0 and up']]

looks like its fine

In [99]:
Appleheader

['id',
 'track_name',
 'size_bytes',
 'currency',
 'price',
 'rating_count_tot',
 'rating_count_ver',
 'user_rating',
 'user_rating_ver',
 'ver',
 'cont_rating',
 'prime_genre',
 'sup_devices.num',
 'ipadSc_urls.num',
 'lang.num',
 'vpp_lic']

In [102]:
free_ios_apps = []
paid_ios_apps = []
for app in ios_eng:
    price = float(app[4])
    if price == 0.0:
        free_ios_apps.append(app)
    else:
        paid_ios_apps.append(app)
        
print('The number of free apps in the apple store data set is: ' + str(len(free_ios_apps)), 
      'and the number of paid apps in the apple store data set is: ' + str(len(paid_ios_apps)))

The number of free apps in the apple store data set is: 3222 and the number of paid apps in the apple store data set is: 2961


so we've filtered out english free apps on both stores. im going to store them in new lists and continue analysis from there

In order to choose the best type of app to build, we need to figure out what type of app is susccesful/popular on both google and apple store. 
The validation strategy to test it out will follow
1. Build a mvp version of the app and put on the google play store
2. If the app gets a good response from users, then we develop it further 
3. If the app becomes profitable in 6 months, we'll develop the ios version and put it on the apple store. 

to analyse the date, I'm going to create a frequency table of the genres to find out the most common. 


In [103]:
ios_data = free_ios_apps
android_data = free_android_apps

In [104]:
Appleheader
#ios_data[0:2]
#ios genres is index 11 

['id',
 'track_name',
 'size_bytes',
 'currency',
 'price',
 'rating_count_tot',
 'rating_count_ver',
 'user_rating',
 'user_rating_ver',
 'ver',
 'cont_rating',
 'prime_genre',
 'sup_devices.num',
 'ipadSc_urls.num',
 'lang.num',
 'vpp_lic']

In [105]:
googleheader
#android genres is index 9, category is index 1

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

In [106]:
def freq_table (dataset, index):
    dict = {}
    total_num_of_apps = len(dataset)
    for row in dataset:
        column = row[index]
        if column in dict:
            dict[column] += 1
        else:
            dict[column] = 1
            
    percentage_dict   = {}   
    for key in dict:
        dict[key]/= total_num_of_apps  #each key divided by the total 
        percentage = round(dict[key]*100,2)
        percentage_dict[key] = percentage
    
    return percentage_dict


In [107]:
freq_table(ios_data, 11)

{'Social Networking': 3.29,
 'Photo & Video': 4.97,
 'Games': 58.16,
 'Music': 2.05,
 'Reference': 0.56,
 'Health & Fitness': 2.02,
 'Weather': 0.87,
 'Utilities': 2.51,
 'Travel': 1.24,
 'Shopping': 2.61,
 'News': 1.33,
 'Navigation': 0.19,
 'Lifestyle': 1.58,
 'Entertainment': 7.88,
 'Food & Drink': 0.81,
 'Sports': 2.14,
 'Book': 0.43,
 'Finance': 1.12,
 'Education': 3.66,
 'Productivity': 1.74,
 'Business': 0.53,
 'Catalogs': 0.12,
 'Medical': 0.19}

In [108]:
freq_table(android_data, 1)

{'ART_AND_DESIGN': 0.64,
 'AUTO_AND_VEHICLES': 0.93,
 'BEAUTY': 0.6,
 'BOOKS_AND_REFERENCE': 2.14,
 'BUSINESS': 4.59,
 'COMICS': 0.62,
 'COMMUNICATION': 3.24,
 'DATING': 1.86,
 'EDUCATION': 1.16,
 'ENTERTAINMENT': 0.96,
 'EVENTS': 0.71,
 'FINANCE': 3.7,
 'FOOD_AND_DRINK': 1.24,
 'HEALTH_AND_FITNESS': 3.08,
 'HOUSE_AND_HOME': 0.82,
 'LIBRARIES_AND_DEMO': 0.94,
 'LIFESTYLE': 3.9,
 'GAME': 9.73,
 'FAMILY': 18.9,
 'MEDICAL': 3.53,
 'SOCIAL': 2.66,
 'SHOPPING': 2.25,
 'PHOTOGRAPHY': 2.94,
 'SPORTS': 3.4,
 'TRAVEL_AND_LOCAL': 2.34,
 'TOOLS': 8.46,
 'PERSONALIZATION': 3.32,
 'PRODUCTIVITY': 3.89,
 'PARENTING': 0.65,
 'WEATHER': 0.8,
 'VIDEO_PLAYERS': 1.79,
 'NEWS_AND_MAGAZINES': 2.8,
 'MAPS_AND_NAVIGATION': 1.4}

lets arrange that in order, transform the dictionary to a tuple then we'll sort it. 

In [109]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        # print(table_sorted)
        print(entry[1], ':', entry[0])

In [110]:
display_table(ios_data,11) #on genres

Games : 58.16
Entertainment : 7.88
Photo & Video : 4.97
Education : 3.66
Social Networking : 3.29
Shopping : 2.61
Utilities : 2.51
Sports : 2.14
Music : 2.05
Health & Fitness : 2.02
Productivity : 1.74
Lifestyle : 1.58
News : 1.33
Travel : 1.24
Finance : 1.12
Weather : 0.87
Food & Drink : 0.81
Reference : 0.56
Business : 0.53
Book : 0.43
Navigation : 0.19
Medical : 0.19
Catalogs : 0.12




from the frequency table above, we can see that games is the most popular type of free english app with an approximate share of 56%, more than half of the apps. secong top with 8% are entertainment apps. 

Let's check to google play store


In [111]:
display_table(android_data,1) #prime genres

FAMILY : 18.9
GAME : 9.73
TOOLS : 8.46
BUSINESS : 4.59
LIFESTYLE : 3.9
PRODUCTIVITY : 3.89
FINANCE : 3.7
MEDICAL : 3.53
SPORTS : 3.4
PERSONALIZATION : 3.32
COMMUNICATION : 3.24
HEALTH_AND_FITNESS : 3.08
PHOTOGRAPHY : 2.94
NEWS_AND_MAGAZINES : 2.8
SOCIAL : 2.66
TRAVEL_AND_LOCAL : 2.34
SHOPPING : 2.25
BOOKS_AND_REFERENCE : 2.14
DATING : 1.86
VIDEO_PLAYERS : 1.79
MAPS_AND_NAVIGATION : 1.4
FOOD_AND_DRINK : 1.24
EDUCATION : 1.16
ENTERTAINMENT : 0.96
LIBRARIES_AND_DEMO : 0.94
AUTO_AND_VEHICLES : 0.93
HOUSE_AND_HOME : 0.82
WEATHER : 0.8
EVENTS : 0.71
PARENTING : 0.65
ART_AND_DESIGN : 0.64
COMICS : 0.62
BEAUTY : 0.6


The google play store data is a bit more varied selection of app which balanced for entertainment & for practical uses. the top genres are fun apps- 'family' apps with 18%, games with 9.73%, practical apps(tools, business, productivity, finance) have about 23%+

In [112]:
display_table(android_data,9) #prime genres

Tools : 8.45
Entertainment : 6.07
Education : 5.35
Business : 4.59
Productivity : 3.89
Lifestyle : 3.89
Finance : 3.7
Medical : 3.53
Sports : 3.46
Personalization : 3.32
Communication : 3.24
Action : 3.1
Health & Fitness : 3.08
Photography : 2.94
News & Magazines : 2.8
Social : 2.66
Travel & Local : 2.32
Shopping : 2.25
Books & Reference : 2.14
Simulation : 2.04
Dating : 1.86
Arcade : 1.85
Video Players & Editors : 1.77
Casual : 1.76
Maps & Navigation : 1.4
Food & Drink : 1.24
Puzzle : 1.13
Racing : 0.99
Role Playing : 0.94
Libraries & Demo : 0.94
Auto & Vehicles : 0.93
Strategy : 0.9
House & Home : 0.82
Weather : 0.8
Events : 0.71
Adventure : 0.68
Comics : 0.61
Beauty : 0.6
Art & Design : 0.6
Parenting : 0.5
Card : 0.45
Casino : 0.43
Trivia : 0.42
Educational;Education : 0.39
Board : 0.38
Educational : 0.37
Education;Education : 0.34
Word : 0.26
Casual;Pretend Play : 0.24
Music : 0.2
Racing;Action & Adventure : 0.17
Puzzle;Brain Games : 0.17
Entertainment;Music & Video : 0.17
Casual;B

there are no games here. curious
i wont be using the above table because it has multiple entries for some categories. 

I'd like to see some of the apps that are in these categories. 


In [113]:
temp_list = []
for row in ios_data:
    genre = row[11]
    name = row[1]
    if genre == 'Entertainment':
        temp_list.append(name)
print(temp_list[0:10])
    

['Netflix', 'Fandango Movies - Times + Tickets', 'Colorfy: Coloring Book for Adults', 'IMDb Movies & TV - Trailers and Showtimes', 'TRUTH or DARE!!! - FREE', 'Mad Libs', 'Twitch', 'Action Movie FX', 'Voice Changer Plus', 'iFunny :)']


So entertainment apps are movie streaming apps like Netflix & IMDB, activity apps that are'nt necessarily games like colorfy. 

In [114]:
temp_list = []
temp_list2 = []
for row in android_data:
    genre = row[1]
    name = row[0]
    if genre == 'FAMILY':
        temp_list.append(name)
    elif genre == 'TOOLS':
        temp_list2.append(name)
print(temp_list[0:15])
print('\n')
print(temp_list2[0:15])
    

['Jewels Crush- Match 3 Puzzle', 'Coloring & Learn', 'Mahjong', 'Super ABC! Learning games for kids! Preschool apps', 'Toy Pop Cubes', 'Educational Games 4 Kids', 'Candy Pop Story', 'Princess Coloring Book', 'Hello Kitty Nail Salon', 'Candy Smash', 'Happy Fruits Bomb - Cube Blast', 'Princess Adventures Puzzles', 'Kids Educational Game 3 Free', 'Puzzle Kids - Animals Shapes and Jigsaw Puzzles', 'Coloring book moana']


['Google', 'Google Translate', 'Moto Display', 'Motorola Alert', 'Motorola Assist', 'Moto Suggestions ™', 'Moto Voice', 'Calculator', 'Device Help', 'Account Manager', 'myMetro', 'File Manager', 'My Telcel', 'Calculator - free calculator, multi calculator app', 'ASUS Sound Recorder']


it seems like the 'family' genre in the android data actually consists of games too, they're just games that are suitable for kids to play. 

so far, it seems like the intersection for ios and android app data is games, a nice niche might be kid's games however, will there be a restriction on the type of ads you can run on kids games & will that in turn restrict our earning potential?


These tables show us the most frequent apps, now its important to check for the apps with the highest number of users. it would make sense to hypothesise that games are the most popular apps because they have a significant share in app numbers. let's see if thats true. 


In [115]:
#Appleheader

In [116]:
#ios_data[5]

# Most popular apps by genre on the App store

To calculate the popular apps by genre on the app stores, a good was is to use the average number of installs of the app. this info is in the google play store but not in the apple store data. As a substitute for this, we're going to find the average no of user ratings for each genre. since ratings are only by users who have installed the app, the average will provide a close enough representation of the most popular apps. 

In [117]:
ios_genres = freq_table(ios_data, 11)
for genres in ios_genres:
    total = 0       #stores the sum of user ratings (the no of rating, not actual ratings) specific to each genre
    len_genre = 0   #stores the number of apps specific to this genre 
    
    for row in ios_data:
        genre_app = row[11]
        
        if genre_app == genres:
            no_ratings = float(row[5])
            total += no_ratings
            len_genre += 1 
            
    average = round(total/len_genre , 2)  
    print(genres,':', average)
    

    

Social Networking : 71548.35
Photo & Video : 28441.54
Games : 22788.67
Music : 57326.53
Reference : 74942.11
Health & Fitness : 23298.02
Weather : 52279.89
Utilities : 18684.46
Travel : 28243.8
Shopping : 26919.69
News : 21248.02
Navigation : 86090.33
Lifestyle : 16485.76
Entertainment : 14029.83
Food & Drink : 33333.92
Sports : 23008.9
Book : 39758.5
Finance : 31467.94
Education : 7003.98
Productivity : 21028.41
Business : 7491.12
Catalogs : 4004.0
Medical : 612.0


Based on the results here, the most populus genre of apps 'Games' is not the most popular amongst users with an average of 22.7k ratings. Genres such as Navigation, Social networking, Reference, Music and Weather have significantly higher average user ratings.
It is important to note that the likelihood of someone to leave a rating or review is higher when the app is providing a service that depends on properties like accuracy and high quality, such as the Reference & Music genres above. Games may have a high number of users but low ratings because people dont feel the need to comment on how entertaining or not a game app is. 

In light of this, lets look at the average rating data with the % share of each app 

In [118]:
share = display_table(ios_data, 11)

Games : 58.16
Entertainment : 7.88
Photo & Video : 4.97
Education : 3.66
Social Networking : 3.29
Shopping : 2.61
Utilities : 2.51
Sports : 2.14
Music : 2.05
Health & Fitness : 2.02
Productivity : 1.74
Lifestyle : 1.58
News : 1.33
Travel : 1.24
Finance : 1.12
Weather : 0.87
Food & Drink : 0.81
Reference : 0.56
Business : 0.53
Book : 0.43
Navigation : 0.19
Medical : 0.19
Catalogs : 0.12


### **lets compare some of this information to make a suggestion for the type of app that should be made**

Navigation - has the higher average with 86k but a % share of 0.19%

Reference - has a high average with 74k but its only 0.49% of apps on the app store

Music - also has a high average with 57k ratings but has a low percentage share with 1.65% 

Photo & Video - looks like it could be promising because it has a 4% share of the apps with 28k average ratings.

Travel - also seems like a viable option with 28k average ratings howver its share is only 1.38%

Games is a top contender because it has a huge share, 58% with a low average rating of 18.9k 

Entertainment - I wouldnt suggest because while its a relatively big share 8%, it has only 10k ratings 

its a bit difficult making a choice of app. an ideal app should have a a big enough share on the app store but does plenty apps reviews mean more or less competition? I think an app that has a median value in this list with a good app share. 
such as 

Let's delve into some categories further

In [119]:
for row in ios_data:
    if row[11] == 'Reference':
        print(row[1], ':', row[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


In [120]:
for row in ios_data:
    if row[11] == 'Navigation':
        print(row[1], ':', row[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


In [121]:
for row in ios_data:
    if row[11] == 'Photo & Video':
        print(row[1], ':', row[5])

Instagram : 2161558
Snapchat : 323905
YouTube - Watch Videos, Music, and Live Streams : 278166
Pic Collage - Picture Editor & Photo Collage Maker : 123433
Funimate video editor: add cool effects to videos : 123268
musical.ly - your video social network : 105429
Photo Collage Maker & Photo Editor - Live Collage : 93781
Vine Camera : 90355
Google Photos - unlimited photo and video storage : 88742
Flipagram : 79905
Mixgram - Picture Collage Maker - Pic Photo Editor : 54282
Shutterfly: Prints, Photo Books, Cards Made Easy : 51427
Pic Jointer – Photo Collage, Camera Effects Editor : 51330
Color Pop Effects - Photo Editor & Picture Editing : 45320
Photo Grid - photo collage maker & photo editor : 40531
iSwap Faces LITE : 39722
MOLDIV - Photo Editor, Collage & Beauty Camera : 39501
Photo Editor by Aviary : 39501
Photo Lab: Picture Editor, effects & fun face app : 34585
Rookie Cam - Photo Editor & Filter Camera : 33921
FotoRus -Camera & Photo Editor & Pic Collage Maker : 32558
PicsArt Photo St

In [122]:
# for row in ios_data:
  #  if row[11] == 'Education':
   #     print(row[1], ':', row[5])

Based on the analysis above, genres with high average ratings are dominated by one or two apps from big tech companies who will prove very difficult competition for us. if those apps were removed, the averages would reduce. 

It might be good practice to shortlist app middle range of the average user ratings & see if theyre also influenced this way Genres like: Finance, Book etc


## Games

In [127]:
 for row in ios_data:
    if row[11] == 'Games':
        print(row[1], ':', row[5])

Clash of Clans : 2130805
Temple Run : 1724546
Candy Crush Saga : 961794
Angry Birds : 824451
Subway Surfers : 706110
Solitaire : 679055
CSR Racing : 677247
Crossy Road - Endless Arcade Hopper : 669079
Injustice: Gods Among Us : 612532
Hay Day : 567344
PAC-MAN : 508808
DragonVale : 503230
Head Soccer : 481564
Despicable Me: Minion Rush : 464312
The Sims™ FreePlay : 446880
Sonic Dash : 418033
8 Ball Pool™ : 416736
Tiny Tower - Free City Building : 414803
Jetpack Joyride : 405647
Bike Race - Top Motorcycle Racing Games : 405007
Kim Kardashian: Hollywood : 397730
Trivia Crack : 393469
WordBrain : 391401
Sniper 3D Assassin: Shoot to Kill Gun Game : 386521
Flow Free : 373857
Geometry Dash Lite : 370370
▻Sudoku : 359832
Fruit Ninja® : 327025
Pixel Gun 3D : 301182
Temple Run 2 : 295211
My Horse : 293857
Word Cookies! : 287095
Dragon City Mobile : 277268
The Simpsons™: Tapped Out : 274501
Plants vs. Zombies™ 2 : 267394
Clash Royale : 266921
Pokémon GO : 257627
CSR Racing 2 : 257100
Star Wars™: 

Good Knight Story : 6282
Real Boxing 2 ROCKY : 6263
Twin Moons Society™: Hidden Mystery : 6219
Faily Brakes : 6183
LEGO® Juniors Create & Cruise : 6152
Mars: Mars : 6116
Dragon Land : 6079
Battle Camp HD - Collect & Evolve Your Monsters : 6055
Thumb Drift - Furious One Touch Car Racing : 6051
Fish Mania™ : 6038
Cops N Robbers (Jail Break) - Survival Mini Game : 6013
Ball King : 5834
Ketchapp Basketball : 5833
Pokémon Shuffle Mobile : 5805
Bakery Story 2 : 5753
Soda Dungeon : 5717
Sonic Dash 2: Sonic Boom : 5697
Green the Planet : 5660
Nom Cat - Endless feeding frenzy arcade game : 5655
Does not Commute : 5645
Pixel Starships™ : 8Bit MMORPG : 5565
Gods Of Rome : 5526
PKTBALL - Endless Arcade Smash Sport : 5492
Brain Dots - Draw and solve! Brain Training Game : 5469
DRAGON BALL Z DOKKAN BATTLE : 5362
Clockmaker – Mystery Match3 Puzzle : 5246
School of Chaos Online MMORPG : 5237
Baby Care & Dress Up - Love & Have Fun with Babies : 5230
RISK: Global Domination : 5217
Kritika: The White Kni

Harvest Land : 1047
Buddy Toss : 1018
Candy Party: Coin Carnival Dozer : 1017
Really Bad Chess : 1015
Pocket Troops : 1005
Lost in Harmony : 1002
PAC-MAN Pop : 1000
Tap Tap Builder : 999
Emoji Blitz : 999
Monster Tail : 998
Mr. Crab 2 : 989
ROCKY™ : 989
Tropical Twist : 982
Cartoon Wars 3 : 979
Flick Field Goal 17 : 968
Fancy Cats : 966
Motor World: Bike Factory : 960
Forsaken World Mobile MMORPG : 952
Color 6 : 951
Disney Magical Dice : 947
RAD Boarding : 944
The Higher Lower Game : 924
Bubble Shoot Pet : 914
Splat Wars : 913
Fitness Girl - Dance and Play at the Gym : 913
Nekosan : 905
Simulator Run Real Hologram : 903
Doodle Jump Easter Special : 900
Robbery Bob 2: Double Trouble : 899
Teen Titans Go Arcade : 897
Leveled! : 897
Puzzle Monster Quest - New MultiPlayer : 894
Ookujira - Giant Whale Rampage : 878
Exit Zone : 875
Duel Master - Best Duelist : 871
Mushroom Wars 2 : 871
Finger Hero: Avoid Obstacles : 870
Water Bottle Flip Challenge : 868
Combo Quest 2 : 855
Clashy Colors : 85

Stop Curry : 250
Supermodel Star - Rule the Runway : 250
SBK16 - Official Mobile Game : 249
High Dive : 249
Osmo Masterpiece : 247
Chaos Centurions : 244
Dropple : 243
Sloomy : 241
VR HORROR : 241
LEGO® Ninjago: Skybound : 239
Sniper: Traffic Hunter : 239
Tap the Blocks : 238
Dawnbringer : 238
Garfield: My BIG FAT Diet : 236
VR Street Jump for Google Cardboard : 236
My Teacher - School Classroom Play & Learn : 234
Chrome Death : 234
EXORUN : 234
Bonecrusher: Free Awesome Endless Skull & Bone Game : 233
Green the Planet 2 : 233
Icarus - A Star's Journey : 230
My Newborn Kitty - Fluffy Care : 230
Flip : 230
TALES OF LINK : 229
Ice Road Trucker Parking Simulator 2 a Real Monster Truck Car Park Racing Game : 228
Crazy Freekick : 228
Logic Traces : 228
Christmas Stories: The Gift of the Magi : 227
Maze Walk VR - Virtual Reality Game Puzzle Apps : 227
Wizard Swipe : 226
■ Square it! : 226
Chef's Quest : 225
Jumping Balls! : 225
Crossy Maze : 223
Las Vegas Valet Limo and Sports Car Parking : 

Micromon : 56
Brio - Don’t Fall! : 56
Escape Game: Forgotten : 55
Dominocity : 55
Donuts Maker Salon : 55
Bouncy Kingdom : 55
ZenDots 2 - One Dot’s Journey : 54
Sweet Princess Beauty Salon : 54
Oops You Died : 54
Diced - A Simple Puzzle Dice Game : 53
Perfect Fit : 53
Color 6: Blitz : 53
Piggy Show : 53
Super Candy: Let's Fix it! : 53
Osmo Numbers : 52
Osmo Monster : 52
Hexy! : 51
Target Acquired : 50
Teeter – Endless Arcade Balancer : 50
Toast Girl : 50
Five Finger Fury : 49
Lock The Block : 49
Escape Game: Relief : 49
Top Tier : 48
What Jump - Free Pop Arcade Game : 48
Ball Escape! : 48
Dot Empires : 47
Stickman Cubed : 47
Spirit Rush : 47
Valley of Fear Virtual Reality : 46
VR Ball for Google Cardboard Virtual Reality : 46
Princess Face Paint - Girls Makeup, Dressup and Makeover Games : 45
Chunky Tanks : 44
Extreme Heavy Trucker Parking Simulator : 44
Ridiculous Parking Simulator a Real Crazy Multi Car Driving Racing Game : 43
Zombies Chasing Me : 43
Curvulate : 43
Water Bottle Flip

## Book

In [128]:
for row in ios_data:
    if row[11] == 'Book':
        print(row[1], ':', row[5])

Kindle – Read eBooks, Magazines & Textbooks : 252076
Audible – audio books, original series & podcasts : 105274
Color Therapy Adult Coloring Book for Adults : 84062
OverDrive – Library eBooks and Audiobooks : 65450
HOOKED - Chat Stories : 47829
BookShout: Read eBooks & Track Your Reading Goals : 879
Dr. Seuss Treasury — 50 best kids books : 451
Green Riding Hood : 392
Weirdwood Manor : 197
MangaZERO - comic reader : 9
ikouhoushi : 0
MangaTiara - love comic reader : 0
謎解き : 0
謎解き2016 : 0


## Finance

In [124]:
for row in ios_data:
    if row[11] == 'Finance':
        print(row[1], ':', row[5])

Chase Mobile℠ : 233270
Mint: Personal Finance, Budget, Bills & Money : 232940
Bank of America - Mobile Banking : 119773
PayPal - Send and request money safely : 119487
Credit Karma: Free Credit Scores, Reports & Alerts : 101679
Capital One Mobile : 56110
Citi Mobile® : 48822
Wells Fargo Mobile : 43064
Chase Mobile : 34322
Square Cash - Send Money for Free : 23775
Capital One for iPad : 21858
Venmo : 21090
USAA Mobile : 19946
TaxCaster – Free tax refund calculator : 17516
Amex Mobile : 11421
TurboTax Tax Return App - File 2016 income taxes : 9635
Bank of America - Mobile Banking for iPad : 7569
Wells Fargo for iPad : 2207
Stash Invest: Investing & Financial Education : 1655
Digit: Save Money Without Thinking About It : 1506
IRS2Go : 1329
Capital One CreditWise - Credit score and report : 1019
U by BB&T : 790
Paribus - Rebates When Prices Drop : 768
KeyBank Mobile : 623
VyStar Mobile Banking for iPhone : 434
Sparkasse - Your mobile branch : 77
VyStar Mobile Banking for iPad : 57
Zaim : 4

Most finance apps with high ratings are app of financial institutions & I dont think running ads would be a good idea

## Food & Drink

In [125]:
for row in ios_data:
    if row[11] == 'Food & Drink':
        print(row[1], ':', row[5])

Starbucks : 303856
Domino's Pizza USA : 258624
OpenTable - Restaurant Reservations : 113936
Allrecipes Dinner Spinner : 109349
DoorDash - Food Delivery : 25947
UberEATS: Uber for Food Delivery : 17865
Postmates - Food Delivery, Faster : 9519
Dunkin' Donuts - Get Offers, Coupons & Rewards : 9068
Chick-fil-A : 5665
McDonald's : 4050
Deliveroo: Restaurant Delivery - Order Food Nearby : 1702
SONIC Drive-In : 1645
Nowait Guest : 1625
7-Eleven, Inc. : 1356
Outback : 805
Bon Appetit : 750
Starbucks Keyboard : 457
Whataburger : 197
Delish Eatmoji Keyboard : 154
Lieferheld - Delicious food delivery service : 29
Lieferando.de : 29
McDo France : 22
Chefkoch - Rezepte, Kochen, Backen & Kochbuch : 20
Youmiam : 9
Marmiton Twist : 2
Open Food Facts : 1


successful apps in the food & drink genre are mostly restaurants. We arent interested in the logistics of all that

## Health and Fitness

In [126]:
for row in ios_data:
    if row[11] == 'Health & Fitness':
        print(row[1], ':', row[5])

Calorie Counter & Diet Tracker by MyFitnessPal : 507706
Lose It! – Weight Loss Program and Calorie Counter : 373835
Weight Watchers : 136833
Sleep Cycle alarm clock : 104539
Fitbit : 90496
Period Tracker Lite : 53620
Nike+ Training Club - Workouts & Fitness Plans : 33969
Plant Nanny - Water Reminder with Cute Plants : 27421
Sworkit - Custom Workouts for Exercise & Fitness : 16819
Clue Period Tracker: Period & Ovulation Tracker : 13436
Headspace : 12819
Fooducate - Lose Weight, Eat Healthy,Get Motivated : 11875
Runtastic Running, Jogging and Walking Tracker : 10298
WebMD for iPad : 9142
8fit - Workouts, meal plans and personal trainer : 8730
Garmin Connect™ Mobile : 8341
Record by Under Armour, connects with UA HealthBox : 7754
Fitstar Personal Trainer : 7496
My Cycles Period and Ovulation Tracker : 7469
Seven - 7 Minute Workout Training Challenge : 6808
RUNNING for weight loss: workout & meal plans : 6407
Lifesum – Inspiring healthy lifestyle app : 5795
Waterlogged - Daily Hydration Tr

Health and Fitness looks like a good contender, it isnt over saturated and people are always looking to lose weight and track their diets. maybe we create an app that combines all the aspects of the top apps in this category. We can integrate calorie counting, meal tracking and and optional period tracking feature for women as women's eating habits often change when theyre more hormonal. We can also run all types of targeted ads for wellness and health to health - focused users. Lets see if its a good fit in the google play store. 
 

## Now lets look through the google play store

In [129]:
googleheader #checking which index installs is in. index 5

['App',
 'Category',
 'Rating',
 'Reviews',
 'Size',
 'Installs',
 'Type',
 'Price',
 'Content Rating',
 'Genres',
 'Last Updated',
 'Current Ver',
 'Android Ver']

In [130]:
display_table(android_data, 5)

1,000,000+ : 15.73
100,000+ : 11.55
10,000,000+ : 10.55
10,000+ : 10.2
1,000+ : 8.39
100+ : 6.92
5,000,000+ : 6.83
500,000+ : 5.56
50,000+ : 4.77
5,000+ : 4.51
10+ : 3.54
500+ : 3.25
50,000,000+ : 2.3
100,000,000+ : 2.13
50+ : 1.92
5+ : 0.79
1+ : 0.51
500,000,000+ : 0.27
1,000,000,000+ : 0.23
0+ : 0.05


In [131]:
Category = freq_table(android_data,1)

for category in Category:
    total = 0
    len_category = 0 
    for row in android_data:
        category_app = row[1]
        if category_app == category:
            installs = row[5]
            
            installs = installs.replace('+', '')
            installs = installs.replace(',', '')
            
            installs = float(installs)
            
            
            total += installs
            total = round(total)
            
            len_category += 1
            
    average_ad = total/len_category
    print(category, ':', average_ad)
    
            
            
            

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3697848.1731343283
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

### Health and fitness seem like it has a high average rating. with 4.1 million installs. lets explore that and some others to see whats going on 

## Communication

In [132]:
for row in android_data:
    if row[1] == 'COMMUNICATION':
        print(row[0], ':', row[5])

WhatsApp Messenger : 1,000,000,000+
Messenger for SMS : 10,000,000+
My Tele2 : 5,000,000+
imo beta free calls and text : 100,000,000+
Contacts : 50,000,000+
Call Free – Free Call : 5,000,000+
Web Browser & Explorer : 5,000,000+
Browser 4G : 10,000,000+
MegaFon Dashboard : 10,000,000+
ZenUI Dialer & Contacts : 10,000,000+
Cricket Visual Voicemail : 10,000,000+
TracFone My Account : 1,000,000+
Xperia Link™ : 10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard : 10,000,000+
Skype Lite - Free Video Call & Chat : 5,000,000+
My magenta : 1,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Seznam.cz : 1,000,000+
Antillean Gold Telegram (original version) : 100,000+
AT&T Visual Voicemail : 10,000,000+
GMX Mail : 10,000,000+
Omlet Chat : 10,000,000+
My Vodacom SA : 5,000,000+
Microsoft Edge : 5,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Calls & Text by Mo+ : 5,000,000+
free 

these are mostly texting apps which were not really interested in because of the privacy required.
Beyond the scope of the task.


## Books and Reference

In [133]:
for row in android_data:
    if row[1] == 'BOOKS_AND_REFERENCE':
        print(row[0], ':', row[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

There seems to be a wide range of apps for books & refence. a good back up option if the health app doesnt work

## Health and Fitness

In [134]:
for row in android_data:
    if row[1] == 'HEALTH_AND_FITNESS' and (row[5] == '1,000,000,000+'  # Checking to apps that a dominant and popular in this category
                                            or row[5] == '500,000,000+'
                                            or row[5] == '100,000,000+'
                                            or row[5] == '10,000,000+'):
        print(row[0], ':', row[5])

Six Pack in 30 Days - Abs Workout : 10,000,000+
Lose Weight in 30 Days : 10,000,000+
Pedometer : 10,000,000+
LG Health : 10,000,000+
Step Counter - Pedometer Free & Calorie Counter : 10,000,000+
Pedometer, Step Counter & Weight Loss Tracker App : 10,000,000+
30 Day Fitness Challenge - Workout at Home : 10,000,000+
Fitbit : 10,000,000+
Garmin Connect™ : 10,000,000+
Runkeeper - GPS Track Run Walk : 10,000,000+
8fit Workouts & Meal Planner : 10,000,000+
Strava Training: Track Running, Cycling & Swimming : 10,000,000+
Workout Trainer: fitness coach : 10,000,000+
7 Minute Workout : 10,000,000+
Freeletics: Personal Trainer & Fitness Workouts : 10,000,000+
Daily Workouts - Exercise Fitness Routine Trainer : 10,000,000+
Google Fit - Fitness Tracking : 10,000,000+
Headspace: Meditation & Mindfulness : 10,000,000+
Period Tracker : 10,000,000+
Period Tracker - Period Calendar Ovulation Tracker : 100,000,000+
Period Tracker Clue: Period and Ovulation Tracker : 10,000,000+
Water Drink Reminder : 10

there not too many dominant apps in this field so it seems to be a good option

# CONCLUSION

In this project, we analysed app data from the google and apple playstore to determine the most effective / free app to build on the platforms

the reccomendation is a health focused app that aids with calorie counting, meal tracking as well as period and ovulation tracking. 